src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -193,64 +251,197 @@ We support
+                        {
+                            enabledFor.Add(new FeatureFilterConfiguration()
+                            {
+                                Name = section[AzureAppConfigurationFeatureFlagFields.Name],
+                                Parameters = new ConfigurationWrapper(section.GetSection(AzureAppConfigurationFeatureFlagFields.Parameters))
+                            });
+                        }
+                    }
+                }
+                else
+                {
+                    enabledFor.Add(new FeatureFilterConfiguration
+                    {
+                        Name = "AlwaysOn"
+                    });
+                }
+            }
+
+            return new FeatureDefinition()
+            {
+                Name = featureName,
+                EnabledFor = enabledFor,
+                RequirementType = requirementType
+            };
+        }
+
+        private string GetFeatureFlagSectionName(IConfigurationSection section)
+        {
+            if (_azureAppConfigurationFeatureFlagSchemaEnabled)
             {
-                return _configuration.GetChildren();
+                return section[AzureAppConfigurationFeatureFlagFields.Id];
             }
 
-            Logger?.LogDebug($"No configuration section named '{FeatureManagementSectionName}' was found.");
+            return section.Key;
+        }
+
+        private IEnumerable<IConfigurationSection> GetFeatureDefinitionSections()
+        {
+            if (!_configuration.GetChildren().Any())
+            {
+                return Enumerable.Empty<IConfigurationSection>();
+            }
+
+            bool hasFeatureManagementSection = _configuration.GetChildren()
+                .Any(section => string.Equals(section.Key, ConfigurationFields.FeatureManagementSectionName, StringComparison.OrdinalIgnoreCase));
+
+            if (!hasFeatureManagementSection && !RootConfigurationFallbackEnabled)
+            {
+                return Enumerable.Empty<IConfigurationSection>();
+            }
+
+            IConfiguration featureManagementConfigurationSection = hasFeatureManagementSection ?
+                _configuration.GetSection(ConfigurationFields.FeatureManagementSectionName) :
+                _configuration;
+
+            if (_azureAppConfigurationFeatureFlagSchemaEnabled)
+            {
+                IConfigurationSection featureFlagsSection = featureManagementConfigurationSection.GetSection(AzureAppConfigurationFeatureFlagFields.FeatureFlagsSectionName);
+
+                return featureFlagsSection.GetChildren();
+            }
+
+            return featureManagementConfigurationSection.GetChildren();
+        }
+
+        private static bool HasAzureAppConfigurationFeatureFlagSchema(IConfiguration featureManagementConfiguration)
+        {
+            bool hasFeatureFlagsSection = featureManagementConfiguration.GetChildren()
```

Comment:
A few suggestions in this section. Get the section while checking for it's existence. Check if feature flags section has any children while doing an all check.

```
            IConfigurationSection featureFlagsConfigurationSection = featureManagementConfiguration
                    .GetChildren()
                    .FirstOrDefault(section => 
                        string.Equals(
                            section.Key,
                            AzureAppConfigurationFeatureFlagFields.FeatureFlagsSectionName,
                            StringComparison.OrdinalIgnoreCase));

            if (featureFlagsConfigurationSection != null)
            {
                if (!string.IsNullOrEmpty(featureFlagsConfigurationSection.Value))
                {
                    return false;
                }

                featureFlagChildren = featureFlagsConfigurationSection.GetChildren();

                return featureFlagChildren.Any() && featureFlagChildren.All(section => int.TryParse(section.Key, out int _));
            }

            return false;
```